components/admin_console/group_settings/group_details/group_details.jsx

```diff
@@ -115,14 +132,59 @@ export default class GroupDetails extends React.PureComponent {
         this.setState({loadingTeamsAndChannels: false});
     }
 
-    onToggle = async (allowReference) => {
-        this.setState({allowReference});
-        this.props.actions.patchGroup(this.props.groupID, {allow_reference: allowReference});
+    onMentionToggle = async (allowReference) => {
+        const {group} = this.props;
+        const isOriginalAllowReference = group.allow_reference;
+
+        this.setState({
+            saveNeeded: true,
+            allowReference,
+            hasAllowReferenceChanged: allowReference !== isOriginalAllowReference}
+        );
+    }
+
+    onMentionChange = async (e) => {
+        const {group} = this.props;
+        const isOriginalGroupMentionName = group.name;
```

Comment:
I would prefer if this was just named `originalGroupMentionName`.

Reply:
sure.